### PR TITLE
Fix for UnflattenFromJSON incorrect handling of empty arrays (CAR 716604)

### PR DESCRIPTION
**Pull request by spathiwa**

### CODE REVIEW

File: make-it/cpplint/Makefile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-LINTARGS = --linelength=160 --filter=-build/header_guard,-readability/casting,-whitespace/braces
```
Comment (mehagar) [inline make-it/cpplint/Makefile]: Thanks for this.

File: source/core/TDCodecVia.cpp [hunk 2]
```diff
@@ -783,7 +783,7 @@ TypeRef TDViaParser::ParseDefaultValue(Boolean mutableValue)
     return cdt;
 }
 //------------------------------------------------------------
-Boolean TDViaParser::PreParseElements(Int32 rank, ArrayDimensionVector dimensionLengths)
+Boolean TDViaParser::PreParseElements(Int32 rank, ArrayDimensionVector dimensionLengths, Int32 *reachedDepth /*= NULL*/)
```
Comment (mehagar) [inline source/core/TDCodecVia.cpp]: Could you use nullptr here?

File: source/include/TDCodecVia.h [hunk 3]
```diff
@@ -121,7 +121,7 @@ class TDViaParser
     Boolean EatJSONPath(SubString* path);
     NIError ParseREPL();
     TypeRef ParseEnqueue();
-    Boolean PreParseElements(Int32 rank, ArrayDimensionVector dimensionLengths);
+    Boolean PreParseElements(Int32 rank, ArrayDimensionVector dimensionLengths, Int32 *reachedDepth = NULL);
```
Comment (mehagar) [inline source/include/TDCodecVia.h]: Use nullptr here instead of NULL.

File: source/core/TDCodecVia.cpp (hunk 2 above)
Comment (spathiwa) [reply source/core/TDCodecVia.cpp]: Sure, done.

File: source/include/TDCodecVia.h (hunk 3 above)
Comment (spathiwa) [reply source/include/TDCodecVia.h]: 👍

File: source/core/TDCodecVia.cpp [hunk 4]
```diff
@@ -804,6 +804,8 @@ Boolean TDViaParser::PreParseElements(Int32 rank, ArrayDimensionVector dimension
     // The opening array_pre like "(" has been parsed before this function has been called.
     Int32 dimIndex;
     while (depth >= 0) {
```
[the remark this answers is not in the document]
Comment (spathiwa) [reply source/core/TDCodecVia.cpp]: It doesn't depend on the caller initializing it.  The while loop already always runs at least once.

[the remark this answers is not in the document]
Comment (spathiwa) [reply source/core/TDCodecVia.cpp]: Ok, done.

**Pull request merged**